### PR TITLE
osd: vary tick interval +/- 5% to avoid scrub livelocks

### PR DESCRIPTION
**Pull request by liewegas**
http://tracker.ceph.com/issues/26890

### CODE REVIEW

File: src/osd/OSD.cc [hunk 1]
```diff
@@ -1978,6 +1976,13 @@ OSD::~OSD()
   delete store;
 }
 
+double OSD::get_tick_interval()
+{
+  // vary +/- 5% to avoid scrub scheduling livelocks
+  double v = OSD_TICK_INTERVAL / 100.0 * (double)(rand() % 10);
```
Comment (xiexingguo) [inline src/osd/OSD.cc]: you might need to do  `(double)(rand() % 11)` for +5%

File: src/osd/OSD.h [hunk 2]
```diff
@@ -1237,7 +1237,8 @@ class OSD : public Dispatcher,
 
 protected:
 
-  static const double OSD_TICK_INTERVAL; // tick interval for tick_timer and tick_timer_without_osd_lock
+  const double OSD_TICK_INTERVAL = { 1.0 };
```
Comment (xiexingguo) [inline src/osd/OSD.h]: Can we make this configurable? Since it can do no harm and  is more elastic...

Comment (tchaikov) [inline src/osd/OSD.h]: could keep this member variable static, like
```c++
static constexpr double OSD_TICK_INTERVAL = { 1.0 };
```

File: src/osd/OSD.h [hunk 3]
```diff
@@ -1237,7 +1237,8 @@ class OSD : public Dispatcher,
...
+  double get_tick_interval();
```
Comment (tchaikov) [inline src/osd/OSD.h]: might want to mark this method `const` if it's not `static`.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2018-08-11_05:04:56-rados-wip-kefu-testing-2018-08-10-1333-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2018-08-12_06:11:58-rados-wip-kefu-testing-2018-08-10-1333-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2018-08-12_07:30:12-rados-wip-kefu-testing-2018-08-10-1333-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2018-08-12_09:15:25-rados-wip-kefu-testing-2018-08-10-1333-distro-basic-mira/

File: src/osd/OSD.cc [hunk 4]
```diff
@@ -1978,6 +1977,14 @@ OSD::~OSD()
   delete store;
 }
 
+double OSD::get_tick_interval()
```
Comment (tchaikov) [inline src/osd/OSD.cc]: missing `const`.

**Pull request merged**